impeller/renderer/backend/gles/sampler_gles.cc

```diff
@@ -21,13 +21,6 @@ bool SamplerGLES::IsValid() const {
 
 static GLint ToParam(MinMagFilter minmag_filter, MipFilter mip_filter) {
   switch (mip_filter) {
-    case MipFilter::kNone:
```

Comment:
I think drawing with `GL_[NEAREST|LINEAR]_MIPMAP_[NEAREST|LINEAR]` sampler will fail in GLES if the mip level is 0. We'll need to set either `GL_NEAREST` or `GL_LINEAR` when the mip count is 1.

Reply:
Please verify that this doesn't break the GLES playgrounds if landing.

Reply:
we can retain this behavior for now without the enum variant. Make the mip_filter optional and treat nullopt as if it were none.

Reply:
This still needs to be addressed for the min filter. Right now, if a texture has a mip count of 1 (as all subpass and filter-created textures do), it will get a min filter that includes mipmap sampling. This will fail to render on some drivers: 
![image](https://user-images.githubusercontent.com/919017/226485385-57d569f1-b5c5-4605-88cb-bf0a332b0830.png)

Reply:
This still needs to be addressed for the min filter. Right now, if a texture has a mip count of 1 (as all subpass and filter-created textures do), it will get a min filter that includes mipmap sampling. This will fail to render on some drivers: 
![image](https://user-images.githubusercontent.com/919017/226485385-57d569f1-b5c5-4605-88cb-bf0a332b0830.png)

Reply:
ahh yeah, got it

Reply:
The test above is a capture of `Play/DisplayListTest.CanDrawBackdropFilter/OpenGLES`, which was rendering anything a couple of days ago prior to this fix: https://github.com/flutter/engine/pull/40425/files